### PR TITLE
Updates CI by adding golangci-lint and yamllint checks

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: test/presubmit-tests.sh [hunk 1]
```diff
@@ -85,10 +85,23 @@ api-unittest(){
...
   go mod vendor
-  go test -p 1 -v ./pkg/... && 
+  go test -p 1 -v ./pkg/... &&
   go test -p 1 -v ./v1/service/...
 }
 
+api-golangci-lint() {
+  info Running go lint
+
+  go mod vendor
```
Comment (SM43) [inline test/presubmit-tests.sh:95]: is go mod vendor required again?  `api-unittest` is ran before this and that has go mod vendor.

File: test/presubmit-tests.sh [hunk 2]
```diff
@@ -85,10 +85,23 @@ api-unittest(){
...
+  golangci-lint run ./pkg/... ./v1/service/...
+}
+
+yaml-lint() {
+  info Running Yamllint
+
+  yamllint -c .yamllint ./config.yaml ./api ./config
```
Comment (SM43) [inline test/presubmit-tests.sh:102]: running on `./api` necessary?

Comment (PuneetPunamiya) [reply test/presubmit-tests.sh:102]: Yes we do have manifest for fixtures, don't we need to apply lint check on that ?

**Comment by PuneetPunamiya** [comment]
Closing this pr by updating CI changes in [Lint PR](https://github.com/tektoncd/hub/pull/199)
/close

**Pull request closed without merging**